### DISCUSSION TITLE
Why are binaries not distributed with the project ?

### DISCUSSION THREAD

**Discussion by dphaldes** [microsoft/build-server-for-gradle#195] [Q&A] [answered]
The releases only contain source code and no binaries. There aren't build instructions in the readme either. GitHub doesn't seem to have a way to mirror a repository and run a build workflow on each release.
It makes using this project very hard

**Comment by jdneo** [comment]
After running `gradle build -x test`, you will find all the required output files under `./server/build/libs`.

![image](https://github.com/user-attachments/assets/bbd4b995-4ed3-41f6-ada9-08c16b8b2b6e)

- plugins: the customized gradle plugin which is used by gradle build server to fetch the project structure information
- runtime: runtime jars used by gradle build server
- server.jar: the gradle build server itself

**Reply by dphaldes** [reply]
Thanks for the reply, but that doesn't answer my question. Why are binaries not distributed for releases?

**Reply by jdneo** [reply]
We have an issue to track this: https://github.com/microsoft/build-server-for-gradle/issues/129

Currently, apart from the vscode-gradle extension, no other clients are using it afaik. When more clients start using it and more people request this, we will consider distributing it somewhere.

**Reply by dphaldes** [reply]
Can this not be used in standalone mode? For clients that directly want to use it like `bsp.nvim` or any other bsp client. 
(I currently want the binaries so that it can be easily be fetched using mason.nvim)

**Reply by dphaldes** [reply]
I would like to bring this topic again. 
> Currently, apart from the vscode-gradle extension, no other clients are using it afaik. When more clients start using it and more people request this, we will consider distributing it somewhere.

Neovim can already use it (with plugins) but you need to manually build the binaries. 
You don't need to ship binaries in a special place either. You just need to make the binaries available through a zip file on you github releases. It can probably be automated using GitHub releases (I will send in a PR if you approve :) ) Maven releases will just get in the way of trying to fetch binaries.

**Reply by jdneo** [reply, marked as answer]
Thank you @dphaldes,

Put binaries to GitHub Release looks fine to me. I think I can manually upload the jars. Because the published jar should to be signed, and the signing task only available in Azure DevOps.

**Reply by dphaldes** [reply]
> Put binaries to GitHub Release looks fine to me. I think I can manually upload the jars

Appreciated! Thank you very much!

**Reply by dphaldes** [reply]
Is there going to be a new release anytime soon? If not, can you please update 0.3.0 with jars? Thank you!
